File: 4.Authentication and authorization for backend engineers/code/python/08_stateless_authentication_7.py
def require_jwt(view):
    @wraps(view)
    def wrapper(*args, **kwargs):
        h = request.headers.get("Authorization", "")
        if not h.startswith("Bearer "):
            return jsonify(error="authentication failed"), 401
        try:
            claims = verify(h.removeprefix("Bearer "))
        except jwt.PyJWTError:
            return jsonify(error="authentication failed"), 401
        # no store lookup ,  identity comes from the verified token
        g.user = {"id": claims["sub"], "role": claims["role"]}
        return view(*args, **kwargs)
    return wrapper

File: 4.Authentication and authorization for backend engineers/code/go/08_stateless_authentication_7.go
import "strings"

func RequireJWT(next http.HandlerFunc) http.HandlerFunc {
    return func(w http.ResponseWriter, r *http.Request) {
        h := r.Header.Get("Authorization")
        if !strings.HasPrefix(h, "Bearer ") {
            http.Error(w, "authentication failed", http.StatusUnauthorized)
            return
        }
        claims, err := Verify(strings.TrimPrefix(h, "Bearer "))
        if err != nil {
            http.Error(w, "authentication failed", http.StatusUnauthorized)
            return
        }
        // no store lookup ,  identity comes from the verified token
        u := User{ID: claims["sub"].(string), Role: claims["role"].(string)}
        next(w, r.WithContext(context.WithValue(r.Context(), "user", u)))
    }
}
